### PR TITLE
Quicksand refactor

### PR DESCRIPTION
**Pull request by comma**
Moved it into its own file out of desert planet file
It's no longer a structure to prevent weird interactions with structure damage.
It now assumes appearance of whatever turf it was spawned on

### CODE REVIEW

File: code/game/objects/structures/quicksand.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
+
+/obj/effect/quicksand
+	name = "quicksand"
+	desc = "There is no candy at the bottom."
```
Comment (MistakeNot4892) [inline code/game/objects/structures/quicksand.dm:4]: 😔

File: code/game/objects/structures/quicksand.dm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		if(busy)
+			to_chat(user, "<span class='notice'>[buckled_mob] is already getting out, be patient.</span>")
+			return
+		var/delay = 60
+		if(user == buckled_mob)
+			delay *=2
+			user.visible_message(
+				"<span class='notice'>\The [user] tries to climb out of \the [src].</span>",
+				"<span class='notice'>You begin to pull yourself out of \the [src].</span>",
+				"<span class='notice'>You hear water sloushing.</span>"
```
Comment (MistakeNot4892) [inline code/game/objects/structures/quicksand.dm]: I know this is copypasta but `sloshing`

File: code/game/objects/structures/quicksand.dm [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+				)
+		else
+			user.visible_message(
+				"<span class='notice'>\The [user] begins pulling \the [buckled_mob] out of \the [src].</span>",
+				"<span class='notice'>You begin to pull \the [buckled_mob] out of \the [src].</span>",
+				"<span class='notice'>You hear water sloushing.</span>"
```
Comment (MistakeNot4892) [inline code/game/objects/structures/quicksand.dm]: As above

File: code/game/objects/structures/quicksand.dm [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+/obj/effect/quicksand/buckle_mob(var/mob/L)
+	..()
+	update_icon()
+
+/obj/effect/quicksand/on_update_icon()
+	if(!exposed)
+		return
+	icon_state = "open"
+	cut_overlays()
+	overlays.Cut()
```
Comment (MistakeNot4892) [inline code/game/objects/structures/quicksand.dm]: Redundant with `cut_overlays()` above.

File: code/game/objects/structures/quicksand.dm [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	if(buckled_mob)
+		add_overlay(image(icon,icon_state="overlay",layer=ABOVE_HUMAN_LAYER))
+
+/obj/effect/quicksand/proc/expose()
+	if(exposed)
+		return
+	visible_message("<span class='warning'>The upper crust breaks, exposing the treacherous quicksands underneath!</span>")
```
Comment (MistakeNot4892) [inline code/game/objects/structures/quicksand.dm]: `quicksand` is already plural I think?

**Comment by MistakeNot4892** [comment]
I know we disagree on SPAN_FART stuff but all the naked spans and `0`/`1` instead of `TRUE`/`FALSE` make me sad. Won't block over it tho.

File: code/game/objects/structures/quicksand.dm (hunk 5 above)
Comment (comma) [reply code/game/objects/structures/quicksand.dm]: Probably yeah

File: code/game/objects/structures/quicksand.dm (hunk 4 above)
Comment (comma) [reply code/game/objects/structures/quicksand.dm]: axed

File: code/game/objects/structures/quicksand.dm (hunk 3 above)
Comment (comma) [reply code/game/objects/structures/quicksand.dm]: de-britifed

File: code/game/objects/structures/quicksand.dm (hunk 2 above)
Comment (comma) [reply code/game/objects/structures/quicksand.dm]: de-britifed

File: code/game/objects/structures/quicksand.dm [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+/obj/effect/quicksand/Initialize()
+	. = ..()
+	var/turf/T = get_turf(src)
+	appearance = T.appearance
+
+/obj/effect/quicksand/user_unbuckle_mob(mob/user)
+	if(buckled_mob && !user.stat && !user.restrained())
+		if(busy)
+			to_chat(user, SPAN_NOTICE("[buckled_mob] is already getting out, be patient."))
```
Comment (MistakeNot4892) [inline code/game/objects/structures/quicksand.dm]: `\The`

Comment (comma) [reply code/game/objects/structures/quicksand.dm]: done

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**